components/TextFields/src/MDCTextInputControllerTextFieldBox.m

```diff
@@ -111,45 +130,74 @@ - (void)updatePlaceholder {
   }
 
   UIEdgeInsets textInsets = [self textInsets:UIEdgeInsetsZero];
-  CGFloat underlineYConstant = textInsets.top + [self estimatedTextHeight] +
+  CGFloat underlineBottomConstant = textInsets.top + [self estimatedTextHeight] +
       [self beneathInputPadding];
   // When floating placeholders are turned off, the underline will drift up unless this is set. Even
   // tho it is redundant when floating is on, we just keep it on always for simplicity.
   // Note: This is an issue only on singleline text fields.
-  if (![self.textInput isKindOfClass:[MDCMultilineTextField class]] && !self.underlineY) {
-    self.underlineY = [NSLayoutConstraint constraintWithItem:self.textInput.underline
-                                                   attribute:NSLayoutAttributeBottom
-                                                   relatedBy:NSLayoutRelationEqual
-                                                      toItem:self.textInput
-                                                   attribute:NSLayoutAttributeTop
-                                                  multiplier:1
-                                                    constant:underlineYConstant];
-    self.underlineY.active = YES;
+  if (!self.underlineBottom) {
+    if (![self.textInput isKindOfClass:[MDCMultilineTextField class]]) {
```

Comment:
super-super-nit / consider:
If I am reading an if statement, I usually find it easier to parse
if (Statement) {
} else {
}
-OVER-
if (!Statement) {
} else {
}

Prefer the first clause being the positive / true outcome.  This may not be useful / helpful / applicable.

Reply:
Done! Thank you.